src/bucket/Bucket.h

```diff
@@ -179,7 +181,37 @@ class Bucket : public std::enable_shared_from_this<Bucket>,
     static uint32_t getBucketVersion(std::shared_ptr<Bucket> const& bucket);
     static uint32_t
     getBucketVersion(std::shared_ptr<Bucket const> const& bucket);
-
+    BucketEntryCounters getBucketEntryCounters() const;
     friend class BucketSnapshot;
 };
+
+enum class LedgerEntryTypeAndDurability : uint32_t
+{
+    ACCOUNT = 0,
+    TRUSTLINE = 1,
+    OFFER = 2,
+    DATA = 3,
+    CLAIMABLE_BALANCE = 4,
+    LIQUIDITY_POOL = 5,
+    TEMPORARY_CONTRACT_DATA = 6,
+    PERSISTENT_CONTRACT_DATA = 7,
+    CONTRACT_CODE = 8,
+    CONFIG_SETTING = 9,
+    TTL = 10
+};
+
+struct BucketEntryCounters
+{
```

Comment:
Nit: Since this is a small struct where we're directly accessing the member, drop the m prefix and just say `entryTypeCounts`